server/middlewares/authorized.js

```diff
@@ -0,0 +1,18 @@
+import userRepo from '../repository/userRepository';
+
+/**
+ * This checks whether the user with the userId is authorized
+ * to complete the request.
+ * @param {object} request
+ * @param {object} response
+ * @param {function} next
+ * @returns {function} callback function
+ */
+const authorization = async (request, response, next) => {
+  const data = await userRepo.getUserbyUsername(request.params.username);
+  const id = data ? data.id : null;
+  request.authorized = id === request.userId; // BOOLEAN
```

Comment:
Can this be renamed like so `request.isAuthorized`?

Reply:
I will change that now. Thanks.